pulsar-zookeeper-utils/src/main/java/com/yahoo/pulsar/zookeeper/ZooKeeperCache.java

```diff
@@ -166,6 +166,23 @@ private void invalidateExists(String path) {
         existsCache.invalidate(path);
     }
 
+    public void asyncInvalidate(String path) {
+        if (scheduledExecutor != null) {
```

Comment:
Actually we have [multiple tests](https://github.com/yahoo/pulsar/blob/master/pulsar-zookeeper-utils/src/test/java/com/yahoo/pulsar/zookeeper/ZkBookieRackAffinityMappingTest.java#L80) which passes null for executor to test specific scenario. Also, 
- we do handle [null executor](https://github.com/yahoo/pulsar/blob/master/pulsar-zookeeper-utils/src/main/java/com/yahoo/pulsar/zookeeper/ZooKeeperCache.java#L116) in ZkCache at multiple places
- so, if caller may pass null executor then I think it should be handled.